frontend/ui/elements/menu_keyboard_layout.lua

```diff
@@ -1,11 +1,105 @@
+local CheckButton = require("ui/widget/checkbutton")
 local FFIUtil = require("ffi/util")
+local HorizontalGroup = require("ui/widget/horizontalgroup")
+local HorizontalSpan = require("ui/widget/horizontalspan")
 local Language = require("ui/language")
+local UIManager = require("ui/uimanager")
+local VerticalGroup = require("ui/widget/verticalgroup")
 local VirtualKeyboard = require("ui/widget/virtualkeyboard")
+local _ = require("gettext")
 
-local sub_item_table = {}
+local input_dialog, check_button_bold, check_button_border
+
+local sub_item_table = {
+    {
+        text = _("Keyboard layout"),
+        sub_item_table = {},
+    },
+    {
+        text = _("Keyboard font size"),
+        keep_menu_open = true,
+        callback = function()
+            input_dialog = require("ui/widget/inputdialog"):new{
+                title = _("Keyboard font size"),
+                input = tostring(G_reader_settings:readSetting("keyboard_font_size") or 22),
+                input_hint = "(16 - 30)",
+                buttons = {
+                    {
+                        {
+                            text = _("Close"),
+                            callback = function()
+                                UIManager:close(input_dialog)
+                            end,
+                        },
+                        {
+                            text = _("Apply"),
+                            is_enter_default = true,
+                            callback = function()
+                                local font_size = tonumber(input_dialog:getInputText())
+                                if font_size and font_size >= 16 and font_size <= 30 then
+                                    G_reader_settings:saveSetting("keyboard_font_size", font_size)
+                                    G_reader_settings:saveSetting("keyboard_label_bold", check_button_bold.checked)
+                                    G_reader_settings:saveSetting("keyboard_label_border", check_button_border.checked)
```

Comment:
Welln actually, I'd rather see these 3 totally related settings with the same prefix :)

Either just keyboard_ (keyboard_font_size, keyboard_bold, keyboard_border) if it's not too short (it may)
or keyboard_key_ (keyboard_key_font_size, keyboard_key_bold, keyboard_key_border)
or keyboard_label as it was.

Reply:
Let's do `keyboard_key_`, because otherwise it is not obvious